### PR TITLE
strongly consistent Query info only

### PR DESCRIPTION
**Pull request by flossypurse**

### CODE REVIEW

File: docs/concepts/what-is-a-query.md [hunk 1]
```diff
@@ -21,38 +27,12 @@ Queries are available to expose the internal Workflow Execution state to the ext
   Query handling logic must be **read-only** and cannot change the Workflow Execution state in any way, or contain any blocking code.
   This means that Query handling logic can not spawn Activity Executions.
 
+Queries are strongly consistent and will operate with the last available state.
```
Comment (Spikhalskiy) [inline docs/concepts/what-is-a-query.md:30]: I would try to save a little more of the original information here. Maybe something like
"
Queries are strongly consistent. A strongly consistent Query is guaranteed to return the state, which includes all Events that came in and were confirmed by the returned response before the Query was issued.
"

**Comment by Spikhalskiy** [comment]
I think it would be great if @mfateev can take a look at the change and verify if we don't remove something important here.

File: docs/concepts/what-is-a-query.md (hunk 1 above)
Comment (flossypurse) [reply docs/concepts/what-is-a-query.md]: Considering the feedback - I did a quick rewrite of the whole thing.

LMK what you think -

File: docs/concepts/what-is-a-query.md [hunk 2]
```diff
@@ -8,51 +8,36 @@ tags:
...
-However, there is a guarantee that eventually the Query would return the actual state.
-This is what it means for a query to be eventually consistent.
+Queries are strongly consistent and are guaranteed to return the most recent state.
+This means that the data reflects the state of all confirmed Events that came in before the Query was sent.
+An Event is considered confirmed if the call creating the Event returned success.
+Events that are created while the Query is outstanding may or may not be reflected in the Workflow state the Query result is based on.
+Strongly consistent Queries may have a small amount of latency.
 
-#### Strong consistency
+A Query can carry arguments to specify which data it is requesting, as every Workflow can expose data to multiple types of Queries.
```
Comment (flossypurse) [inline docs/concepts/what-is-a-query.md:31]:
```suggestion
A Query can carry arguments to specify which data it is requesting.
And each Workflow can expose data to multiple types of Queries.
```

File: docs/concepts/what-is-a-query.md [hunk 3]
```diff
@@ -8,51 +8,37 @@ tags:
...
 
-#### Strong consistency
+A Query can carry arguments to specify which data it is requesting.
+And each Workflow can expose data to multiple types of Queries.
 
-Query has another consistency level called strong consistency.
-A strongly consistent Query is guaranteed to return the state, which includes all Events that came before the Query was issued.
-An Event is considered to have come before a Query if the call creating the Event returned success before the Query was issued.
-Events that are created while the Query is outstanding may or may not be reflected in the Workflow state the Query result is based on.
+A Query must never mutate the state of the Workflow Execution – that is, Queries are **read-only** and cannot contain any blocking code.
```
Comment (djmagee) [inline docs/concepts/what-is-a-query.md:34]:
```suggestion
A Query must never mutate the state of the Workflow Execution—that is, Queries are _read-only_ and cannot contain any blocking code.
```
Formatting change per "Italic" section of https://developers.google.com/style/text-formatting.

File: docs/concepts/what-is-a-query.md [hunk 4]
```diff
@@ -8,51 +8,37 @@ tags:
...
+This means that Query handling logic can not schedule Activity Executions for example.
```
Comment (djmagee) [inline docs/concepts/what-is-a-query.md:35]:
```suggestion
This means, for example, that Query handling logic cannot schedule Activity Executions.
```

File: docs/concepts/what-is-a-query.md [hunk 5]
```diff
@@ -8,51 +8,37 @@ tags:
...
 
-**Implementation guides:**
+#### Stack Trace Query
 
-- [How to send a Query to a Workflow Execution in Go](/docs/go/how-to-send-a-query-to-a-workflow-execution-in-go)
-- [How to handle a Query in a Workflow in Go](/docs/go/how-to-handle-a-query-in-a-workflow-in-go)
-- [How to use Queries in Java](/docs/java/queries)
-- [How to use Queries in PHP](/docs/php/queries)
-- [How to send a Query to a Workflow Execution using tctl](/docs/tctl/workflow/query)
+In many SDKs the Temporal Client exposes a predefined `__stack_trace` Query that returns the stack trace of all the threads owned by that Workflow Execution.
```
Comment (djmagee) [inline docs/concepts/what-is-a-query.md:41]:
```suggestion
In many SDKs, the Temporal Client exposes a predefined `__stack_trace` Query that returns the stack trace of all the threads owned by that Workflow Execution.
```

File: docs/concepts/what-is-a-query.md [hunk 6]
```diff
@@ -8,51 +8,37 @@ tags:
...
+This is a great way to troubleshoot a Workflow Execution in production.
+If a Workflow Execution has been stuck at a state for longer than an expected period of time, for example, you can send a `__stack_trace` Query to return the current call stack.
```
Comment (djmagee) [inline docs/concepts/what-is-a-query.md:43]:
```suggestion
For example, if a Workflow Execution has been stuck at a state for longer than an expected period of time, you can send a `__stack_trace` Query to return the current call stack.

```

File: docs/go/how-to-handle-a-query-in-a-workflow-in-go.md [hunk 7]
```diff
@@ -50,3 +50,11 @@ func MyWorkflow(ctx workflow.Context, input string) error {
   return nil
 }
 ```
+
+For example, suppose your query handler function takes 2 parameters:
```
Comment (djmagee) [inline docs/go/how-to-handle-a-query-in-a-workflow-in-go.md:54]:
```suggestion
For example, suppose your query handler function takes two parameters:
```

File: docs/go/how-to-send-a-query-to-a-workflow-execution-in-go.md [hunk 8]
```diff
@@ -5,84 +5,43 @@ sidebar_label: Sending Queries
...
+  // ...
 }
-fmt.Printf("Response: %v\n", response) // "foo" + currentState + "baz"
+// ...
 ```
 
-#### `QueryWorkflowWithOptions()` API
-
-The `QueryWorkflowWithOptions()` function is an alternative way to send a Query to a Workflow that has a few extra configuration options.
+The `QueryWorkflowWithOptions()` API provides similar functionality, but with the ability to set additional configurations through the [QueryWorkflowWithOptionsRequest](https://pkg.go.dev/go.temporal.io/sdk/client#QueryWorkflowWithOptionsRequest).
```
Comment (djmagee) [inline docs/go/how-to-send-a-query-to-a-workflow-execution-in-go.md:33]:
```suggestion
The `QueryWorkflowWithOptions()` API provides similar functionality, but with the ability to set additional configurations through [QueryWorkflowWithOptionsRequest](https://pkg.go.dev/go.temporal.io/sdk/client#QueryWorkflowWithOptionsRequest).
```

File: docs/go/how-to-send-a-query-to-a-workflow-execution-in-go.md [hunk 9]
```diff
@@ -5,84 +5,43 @@ sidebar_label: Sending Queries
...
+When using this API you will also receive a structured [QueryWorkflowWithOptionsResponse](https://pkg.go.dev/go.temporal.io/sdk/client#QueryWorkflowWithOptionsResponse)
```
Comment (djmagee) [inline docs/go/how-to-send-a-query-to-a-workflow-execution-in-go.md:34]:
```suggestion
When using this API, you will also receive a structured response of type [QueryWorkflowWithOptionsResponse](https://pkg.go.dev/go.temporal.io/sdk/client#QueryWorkflowWithOptionsResponse).
```

**Review by djmagee — APPROVED** [review]

File: docs/concepts/what-is-a-query.md [hunk 10]
```diff
@@ -8,51 +8,37 @@ tags:
...
+The Workflow must have a Query handler that is developed to handle that Query and provide data that represents the state of the Workflow Execution.
 
-The Query may or may not return the updated state that quickly.
-However, there is a guarantee that eventually the Query would return the actual state.
-This is what it means for a query to be eventually consistent.
+Queries are strongly consistent and are guaranteed to return the most recent state.
+This means that the data reflects the state of all confirmed Events that came in before the Query was sent.
+An Event is considered confirmed if the call creating the Event returned success.
+Events that are created while the Query is outstanding may or may not be reflected in the Workflow state the Query result is based on.
+Strongly consistent Queries may have a small amount of latency.
```
[the remark this answers is not in the document]
Comment (mfateev) [reply docs/concepts/what-is-a-query.md:29]: I think this phrase doesn't add any value.

**Pull request merged**